### ISSUE TITLE
Simple operations will report errors. eg "log    ${${0.2}-${1}}"

### ISSUE THREAD

**Issue by chengaomin** [robotframework/robotframework#4176]
![image](https://user-images.githubusercontent.com/4586906/147445690-80878fb0-f813-493e-a22f-de9183e4e620.png)

`20211227 15:15:39.701 :  FAIL : Resolving variable '${0.2-1}' failed: SyntaxError: invalid syntax (<string>, line 1)`

**Comment by pekkaklarck** [comment]
You are using the extended variable syntax and it is resolved like this:
1.Try to find variable with name `0.2-1`. Not found.
2. Find base name for a possible variable that is used with the extended syntax. Base name cannot contain special characters. In this case the base name is `0`.
3. Try to find variable with the base name. `${0}` exists and is equal to zero.
4. Evaluate expression in Python so that the base name is replaced with a variable containing the value got earlier. This causes a syntax error also in Python:
   ```python
   >>> var = 0
   >>> var.2 - 1
     File "<stdin>", line 1
       var.2 - 1
          ^
   SyntaxError: invalid syntax
   ```

In other words, the problem here is the `.` character. You could restructure the variable to `${2 / 10 - 1}` to avoid it. Alternatively you could use the newish inline Python evaluation syntax and use `${{0.2 -1}}`. In this case the expression is evaluated directly in Python.
